support-frontend/app/controllers/Application.scala

```diff
@@ -315,11 +308,13 @@ class Application(
 
   private def contributionsHtml(
       countryCode: String,
-      geoData: GeoData,
-      idUser: Option[IdUser],
       campaignCode: Option[String],
-      guestAccountCreationToken: Option[String],
-  )(implicit request: RequestHeader, settings: AllSettings) = {
+  )(implicit request: OptionalAuthRequest[AnyContent], settings: AllSettings) = {
+    val geoData = request.geoData
+    val idUser = request.user
+
+    // This will be present if the token has been flashed into the session by the PayPal redirect endpoint
+    val guestAccountCreationToken = request.flash.get("guestAccountCreationToken")
```

Comment:
I can't say I fully understand why we have `contributionsHtml` and `contributionsLanding` but it's helped here - I just needed to shift some of the arguments into the actual method.